src/selectors/horizontalScrollbarVisible.js

```diff
@@ -8,30 +8,44 @@
  *
  * @providesModule horizontalScrollbarVisible
  */
-import availableViewportWidth from 'availableViewportWidth';
+import Scrollbar from 'Scrollbar';
 import { createSelector } from 'reselect';
-import scrollContentWidth from 'scrollContentWidth';
+import minColWidth from 'minColWidth';
+
+export const ScrollbarState = {
```

Comment:
This is the ternary state of the scrollbar.  Funky approach... but best I could figure

Reply:
IF_VERTICAL_VISIBLE is a bit confusing, esp. where we have an if(horizontalScrollbarVisible === ScrollbarState.IF_VERTICAL_VISIBLE) check below. Maybe just CONDITIONAL instead?

Reply:
Ya, in the follow up review, I change this to IF_SCROLL_Y.  Do you think that's better?  The condition becomes `if (scrollStateX === ScrollbarState.IF_SCROLL_Y)`.  Does that work?

My concern with conditional is it's not very descriptive.  So while it may not come across as weird, it leaves things opaque if you try to dig deeper.

Having the weird if statement may give you a double take, but I've added a comment "Handle case where vertical scrollbar makes horizontal scrollbar necessary" to help explain that.

Reply:
Yeah I just saw that.. and yes CONDITIONAL is still a little vague, maybe SCROLL_Y_PRESENT?  I'm fine with w/e you prefer though, my only gripe was that IF_SCROLL_Y doesn't sound like a "state" like the more natural hidden / visible states.

Reply:
Ya, you're right.  Do you think SCROLL_Y_VISIBLE would be good?  I like that that parallels the VISIBLE state more than present, but then again it's confusing for that reason too...

Maybe it could be something like JOINT_SCROLLBARS or MUTUAL_SCROLL to express the state is either both dimensions scroll or neither?

Reply:
How about BOUND_TO_Y? `if (scrollStateX === ScrollbarState.BOUND_TO_Y)` sounds pretty intuitive to me